html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+namespace Drupal\hpc_api\Query;
+
+/**
+ * Get plan information.
+ */
+class EndpointPlanQuery extends EndpointQuery {
+
+  /**
+   * Get plan.
+   */
+  public function getPlan($plan_id) {
+    $this->setEndpoint('public/plan/id/' . $plan_id);
+
+    $data = $this->getData();
+
+    $plan = [
```

Comment:
If it's preferred, we can also return a (standard) class

Reply:
Ok, I get the idea. That will definitely be of use, I'm just not sure we should merge this until we figure out how we could really share those modules. For the moment I just copied them over from FTS Public, made the necessary fixes to make the linting pass and committed the changes back to the fts-d8-site repo where they await further review by Hardik. But I'd like to prevent them to diverge for the moment.

Should we create another repository for them?

Reply:
Personally, I would return a class instead of an array.